test/ports_test.dart

```diff
@@ -430,85 +430,87 @@ void testSingleResponseChannel() {
-      expect(v, 42);
-    });
-  });
-
-  test("TimeoutOnTimeoutAsync", () {
-    var channel = new SingleResponseChannel(
-        timeout: MS * 100, onTimeout: () => new Future.value(42));
-    return channel.result.then((v) {
-      expect(v, 42);
-    });
-  });
-
-  test("TimeoutOnTimeoutAsyncError", () {
-    var channel = new SingleResponseChannel(
-        timeout: MS * 100, onTimeout: () => new Future.error(42));
-    return channel.result.then((v) {
-      fail("unreachable");
-    }, onError: (v, s) {
-      expect(v, 42);
-    });
-  });
+  group('Timeout', () {
+    test("Timeout", () {
+      var channel = new SingleResponseChannel(timeout: MS * 100);
+      return channel.result.then((v) {
+        expect(v, null);
+      });
+    });
+
+    test("TimeoutThrow", () {
+      var channel =
+          new SingleResponseChannel(timeout: MS * 100, throwOnTimeout: true);
+      return channel.result.then((v) {
+        fail("unreachable");
+      }, onError: (v, s) {
+        expect(v is TimeoutException, isTrue);
+      });
+    });
+
+    test("TimeoutThrowOnTimeoutAndValue", () {
+      var channel = new SingleResponseChannel(
+          timeout: MS * 100,
+          throwOnTimeout: true,
+          onTimeout: () => 42,
+          timeoutValue: 42);
+      return channel.result.then((v) {
+        fail("unreachable");
+      }, onError: (v, s) {
+        expect(v is TimeoutException, isTrue);
+      });
+    });
+
+    test("TimeoutOnTimeout", () {
+      var channel =
+          new SingleResponseChannel(timeout: MS * 100, onTimeout: () => 42);
+      return channel.result.then((v) {
+        expect(v, 42);
+      });
+    });
+
+    test("TimeoutOnTimeoutAndValue", () {
+      var channel = new SingleResponseChannel(
+          timeout: MS * 100, onTimeout: () => 42, timeoutValue: 37);
+      return channel.result.then((v) {
+        expect(v, 42);
+      });
+    });
+
+    test("TimeoutValue", () {
+      var channel =
+          new SingleResponseChannel(timeout: MS * 100, timeoutValue: 42);
+      return channel.result.then((v) {
+        expect(v, 42);
+      });
+    });
+
+    test("TimeoutOnTimeoutError", () {
+      var channel = new SingleResponseChannel(
+          timeout: MS * 100, onTimeout: () => throw 42);
+      return channel.result.then((v) {
+        fail("unreachable");
+      }, onError: (v, s) {
+        expect(v, 42);
+      });
+    });
+
+    test("TimeoutOnTimeoutAsync", () {
+      var channel = new SingleResponseChannel(
+          timeout: MS * 100, onTimeout: () => new Future.value(42));
+      return channel.result.then((v) {
+        expect(v, 42);
+      });
+    });
+
+    test("TimeoutOnTimeoutAsyncError", () {
+      var channel = new SingleResponseChannel(
+          timeout: MS * 100, onTimeout: () => new Future.error(42));
+      return channel.result.then((v) {
+        fail("unreachable");
+      }, onError: (v, s) {
+        expect(v, 42);
+      });
+    });
+  }, tags: 'skip_Dart_v1_x');
```

Comment:
And why does this group fail?
It should work, and there's nothing Dart 1 special about the code.

Reply:
Ditto...